apps/system/js/icc_events.js

```diff
@@ -175,28 +174,21 @@ var icc_events = {
       ' - Events list:', eventList);
 
     var conn = icc.getConnection(message.iccId);
-    this.register_icc_event_voicechange = function(evt) {
-      icc_events.handleLocationStatus(message, evt);
-    };
     conn.removeEventListener('voicechange',
       this.register_icc_event_voicechange);
 
-    this.register_icc_event_idlescreen = function() {
-      icc_events.handleIdleScreenAvailableEvent(message);
-    };
+    if (this.stkUserActivity) {
+      navigator.removeIdleObserver(this.stkUserActivity);
+    }
```

Comment:
When STK_CMD_SET_UP_EVENT_LIST command is received, the whole eventList should be handled again.
If we understand every STK events as a switch, eventList is a list that describes how many events should be turned "ON". It should be recognized as "OFF" if the event is not in the list.
So, all event registrations (turn off all switches) should be removed before checking the events.

The above comment is based on [3GPP TS 11.14, 6.4.16 SET UP EVENT LIST] for your reference. You can see the comment 13 at bug 1088611 as well.
https://bugzilla.mozilla.org/show_bug.cgi?id=1088611#c13

If you think there is any wrong, please tell me. Thank you very much!